system/BaseModel.php

```diff
@@ -1578,7 +1578,7 @@ protected function transformDataToArray($data, string $type): array
         // properties representing the collection elements, we need to grab
         // them as an array.
         if (is_object($data) && ! $data instanceof stdClass) {
-            $data = $this->objectToArray($data, true, true);
+            $data = $this->objectToArray($data, ($type === 'update'), true);
```

Comment:
The parenthesis is not needed.
```suggestion
            $data = $this->objectToArray($data, $type === 'update', true);
```

Reply:
It is difficult to read.

Reply:
Hmm. On second look I guess you're right.